### PR TITLE
change gradle task syntax and placement

### PR DESCRIPTION
**Pull request by munkiki7**
Reason for changes: random android smoke tests failures during CI tasks.

We have noticed that sometimes the upload task is executed before the assemble task is finished, which resulted in not all .so files being generated at the moment it was executed. After that the uploaded symbols validation was failing.

Now we are attaching the symbol upload task to either `assembleDebug` or `assembleRelease` tasks of the launcher build.gradle instead of the root build.gradle file.

One of the other advantages is that with the previous approach the symbol upload task was executed during the gradle sync process in the Android Studio (if the project was exported), as it was bound to the last task being executed (even though it was not a build-related task).

Previous syntax:
```
gradle.taskGraph.whenReady { taskGraph ->
    gradle.taskGraph.allTasks[-1].doLast { task ->
        //task body
    }
}
```
New syntax (needs to be wrapped in the afyterEvaluate call, because we have to wait until the task list is built):
```
afterEvaluate {
    task sentryUploadSymbols {
        doLast {
            //task body
        }
    }
    
    tasks.assembleDebug.finalizedBy sentryUploadSymbols
    tasks.assembleRelease.finalizedBy sentryUploadSymbols
}
```

### CODE REVIEW

File: src/Sentry.Unity.Editor/Android/DebugSymbolUpload.cs [hunk 1]
```diff
@@ -53,8 +54,8 @@ private string _symbolUploadTaskFormat
                     stringBuilder.AppendLine($"        def sentryLogFile = new FileOutputStream('{logsDir}/sentry-symbols-upload.log')");
                 }
                 stringBuilder.AppendLine("        exec {");
-                stringBuilder.AppendLine("            environment 'SENTRY_PROPERTIES', './sentry.properties'");
-                stringBuilder.AppendLine($"            executable '{SentryCliMarker}'");
+                stringBuilder.AppendLine("            environment 'SENTRY_PROPERTIES', file(\"${rootDir}/sentry.properties\").absolutePath");
```
Comment (bitsandfoxes) [inline src/Sentry.Unity.Editor/Android/DebugSymbolUpload.cs:57]: Where is the change to the path coming from?

File: src/Sentry.Unity.Editor/Android/DebugSymbolUpload.cs [hunk 2]
```diff
@@ -81,7 +86,7 @@ private string _symbolUploadTaskFormat
 
             _unityProjectPath = unityProjectPath;
             _gradleProjectPath = gradleProjectPath;
-            _gradleScriptPath = Path.Combine(_gradleProjectPath, "build.gradle");
+            _gradleScriptPath = Path.Combine(_gradleProjectPath, "launcher/build.gradle");
```
Comment (bitsandfoxes) [inline src/Sentry.Unity.Editor/Android/DebugSymbolUpload.cs:89]: Ah, I see. Because the SDK now modifies the `launcher/build.gradle`. Why?

**Comment by bitsandfoxes** [comment]
Thanks for tackling this!

File: src/Sentry.Unity.Editor/Android/DebugSymbolUpload.cs (hunk 2 above)
Comment (munkiki7) [reply src/Sentry.Unity.Editor/Android/DebugSymbolUpload.cs:89]: The task is now added to the topmost project - launcher that depends on unityFramework target.

We are now executing the symbol upload task after the `assembleDebug` or `assembleRelease` tasks, which do not exist in the scope of the `build.gradle` file we were modifying previously.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
 ## Unreleased
 
+- change gradle task syntax and placement ([#1657](https://github.com/getsentry/sentry-unity/pull/1657))
+
 ### Fixes
 
 - The SDK no longer crashes on Android versions 5 and 6 with native support enabled ([#1652](https://github.com/getsentry/sentry-unity/pull/1652))
```
Comment (bitsandfoxes) [inline CHANGELOG.md:9]:
```suggestion
### Fixes

- Updated the symbol upload task on Gradle to be guaranteed to run last ([#1657](https://github.com/getsentry/sentry-unity/pull/1657))
- The SDK no longer crashes on Android versions 5 and 6 with native support enabled ([#1652](https://github.com/getsentry/sentry-unity/pull/1652))
```

File: src/Sentry.Unity.Editor/Android/DebugSymbolUpload.cs (hunk 2 above)
Comment (munkiki7) [reply src/Sentry.Unity.Editor/Android/DebugSymbolUpload.cs:89]: That depends on whether anything else is also using the locations of those files. But we can check.

Comment (bitsandfoxes) [reply src/Sentry.Unity.Editor/Android/DebugSymbolUpload.cs:89]: Sentry CLI is the one looking for the `.properties` so I think it's safe to keep all that in once place.

**Review by bitsandfoxes — APPROVED** [review]

**Pull request merged**